### PR TITLE
Introduce round-robining for scan and ft.search commands

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

**Comment by sazzad16** [comment]
@bsbodden

File: src/main/java/redis/clients/jedis/util/JedisRoundRobinBase.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      }
+    } else {
+      throw new IllegalArgumentException(connection.getValue().getClass() + "is not supported.");
+    }
+
+    lastReply = builder.build(rawReply);
+    iterationCompleted = isIterationCompleted(lastReply);
+    if (iterationCompleted) {
+      if (connections.isEmpty()) {
+        roundRobinCompleted = true;
```
Comment (bsbodden) [inline src/main/java/redis/clients/jedis/util/JedisRoundRobinBase.java:76]: Do we have any potential issues with threading / re-entrant code here?

**Review by bsbodden — COMMENTED** [review]
LGTM! Nice job!

File: src/main/java/redis/clients/jedis/util/JedisRoundRobinBase.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/util/JedisRoundRobinBase.java:76]: @bsbodden I thought it is not necessary to make these round robin objects thread safe. (Initiating them; e.g. calling scan, ftSearch, are thread safe.)

File: src/main/java/redis/clients/jedis/ScanRoundRobin.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import redis.clients.jedis.providers.ConnectionProvider;
+import redis.clients.jedis.resps.ScanResult;
+import redis.clients.jedis.util.JedisRoundRobinBase;
+
+public class ScanRoundRobin extends JedisRoundRobinBase<ScanResult<String>> {
+
+  private final int count;
+  private final Function<String, CommandArguments> args;
+
+  public ScanRoundRobin(ConnectionProvider connectionProvider, int batchCount, String match) {
```
Comment (chayim) [inline src/main/java/redis/clients/jedis/ScanRoundRobin.java:15]: IMHO having the function being named the same as the class feels clunky. The class looks like it should be named RoundRobinScanner - or something, just to differentiate.

I recognize it's the norm (plenty of socket.socket in the universe), but  IMHO purpose based function/class combinations are more clear

**Review by dengliming — APPROVED** [review]

**Review by yangbodong22011 — APPROVED** [review]
LGTM

**Pull request merged**